Gordon360/Services/HousingService.cs

```diff
@@ -31,39 +32,89 @@ public IEnumerable<HousingViewModel> GetAll()
-            result2 = RawSqlQuery<HousingViewModel>.query("GET_AA_APPID_BY_NAME_AND_DATE @NOW, @MODIFIER_ID", timeParam, idParam); //run stored procedure
-            foreach(string applicant in appIds){
-                result3 = RawSqlQuery<HousingAppToSubmitViewModel>.query("INSERT_AA_APPLICANT @APPLICATION_ID, @ID_NUM, @APRT_PROGRAM,  @SESS_CDE", result2, applicant, null, sessionParam); //run stored procedure
-            }
+            int appId = -1;
+
+            if (apartAppId == -1)
+            {
+                // If an application ID was not passed in, then check if an application already exists
+                idResult = RawSqlQuery<ApartmentAppIDViewModel>.query("GET_AA_APPID_BY_STU_ID_AND_SESS @SESS_CDE, @STUDENT_ID", sessionParam, modParam); //run stored procedure
+                if (idResult == null)
+                {
+                    // If an existing application was not found for this modifier, then insert a new application entry in the database
+                    result = RawSqlQuery<ApartmentAppSaveViewModel>.query("INSERT_AA_APPLICATION @NOW, @MODIFIER_ID", timeParam, modParam); //run stored procedure
+                    if (result == null)
+                    {
+                        throw new ResourceNotFoundException() { ExceptionMessage = "The application could not be saved." };
+                    }
+
+                    // The following is ODD, I know. It seems you cannot execute the same query with the same sql parameters twice.
+                    // Thus, these two sql params must be recreated after being used in the last query:
 
+                    timeParam = new SqlParameter("@NOW", now);
+                    modParam = new SqlParameter("@MODIFIER_ID", modifierId);
 
-            if (result == null)
+                    idResult = RawSqlQuery<ApartmentAppIDViewModel>.query("GET_AA_APPID_BY_NAME_AND_DATE @NOW, @MODIFIER_ID", timeParam, modParam); //run stored procedure
+                    if (idResult == null)
+                    {
+                        throw new ResourceNotFoundException() { ExceptionMessage = "The new application ID could not be found." };
+                    }
+                }
+                ApartmentAppIDViewModel idModel = idResult.ElementAt(0);
+
+                appId = idModel.AprtAppID;
+            }
+            else
             {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The data was not found." };
+                // Use the application ID number that was passed in as a parameter
+                appId = apartAppId;
+            }
+            
+            SqlParameter appIdParam = null;
+            SqlParameter idParam = null;
+            SqlParameter programParam = null;
+
+            foreach (string id in applicantIds) {
+                // this constructs new SqlParameters each time we iterate (despite that only 1/4 will actually be different
```

Comment:
I think it would be more clear if you specify which one is being different.

Reply:
I think it would be best to simply say that "the SqlParameters must be re-created before each SQL Query to prevent errors"
Something similar that is short but clear and can be put before each block of "new SqlParameter ..." so that readers do not have to search for the explanation in a separate comment elsewhere in the file.

Reply:
This change has been committed